devops/scripts/boot-strap-venv.sh

```diff
@@ -1,19 +1,39 @@
 #!/bin/bash
-#
+# shellcheck disable=SC1090
 # Shared logic to be sourced for bootstrapping a development virtualenv
 
-function virtualenv_bootstrap() {
-    # If there is no virtualenv started, create one, install deps, source
-    if [ -z "${VIRTUAL_ENV:-}" ]; then
+set -eo pipefail
+
+PYTHON_VERSION="${PYTHON_VERSION:-3}"
+VENV=".venv"
+if [[ $PYTHON_VERSION != "3" ]]
+then
+    VENV=".venv${PYTHON_VERSION}"
+fi
 
-        if [ ! -d ".venv" ]; then
-            virtualenv -p "$(which python2)" .venv
+function virtualenv_bootstrap() {
+    if [ -d "${VENV}" ]
+    then
+        VENV_VERSION=$("$VENV/bin/python" -c "from __future__ import print_function; import sys; print(sys.version_info[0])")
+        if [[ $VENV = ".venv" && $VENV_VERSION == 2 ]]
+        then
+            relo="${VENV}-${VENV_VERSION}-$(date +'%Y%m%d-%H%M%S')"
+            echo "Default virtualenv in .venv is Python 2; renaming it to ${relo}."
+            mv "${VENV}" "${relo}"
         fi
+    fi
 
-        # https://github.com/pypa/virtualenv/issues/1029
-        set +u
-        source .venv/bin/activate
-        set -u
-        pip install -r securedrop/requirements/develop-requirements.txt
+    if [ ! -d "$VENV" ]
+    then
+        p=$(which "python${PYTHON_VERSION}")
+        virtualenv -p "${p}" "${VENV}"
     fi
+
+    "${VENV}/bin/pip" install -q pip==19.1.1
```

Comment:
hmm remind me why we pin this version? (can we add a comment explaining for future us?)

Reply:
It wasn't necessary.